src/inc/til/color.h

```diff
@@ -190,6 +190,31 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 
             return wss.str();
         }
+        static til::color HexToColor(const std::wstring str)
+        {
+            THROW_HR_IF(E_INVALIDARG, str.size() != 9 && str.size() != 7);
+            THROW_HR_IF(E_INVALIDARG, str.at(0) != '#');
+
+            std::wstring workingString = str.substr(1);
+            til::color color;
+
+            // Alpha is specified
+            if (workingString.size() == 8)
+            {
+                color.a = ::base::checked_cast<uint8_t>(std::stoi(workingString.substr(0, 2), nullptr, 16));
+                workingString = workingString.substr(2);
+            }
+            else
+            {
+                color.a = 255;
+            }
+
+            color.r = ::base::checked_cast<uint8_t>(std::stoi(workingString.substr(0, 2), nullptr, 16));
+            color.g = ::base::checked_cast<uint8_t>(std::stoi(workingString.substr(2, 2), nullptr, 16));
+            color.b = ::base::checked_cast<uint8_t>(std::stoi(workingString.substr(4, 2), nullptr, 16));
+
+            return color;

```

Comment:
not the world's biggest fan of having another one of these. is there any way at all we can consolidate, or make it so we don't need to support the `ARGB` hex version?

Reply:
I guess technically the ColorPicker already has its own TextBox where you can enter a hex color. I could just make the TextBox read-only and force users to have to bring up the color picker to enter a hex color? This function is really only used for the ConvertingBack from hex to a color when entered in one of the TextBoxes next to the color picker.

Reply:
This may also help out the observability problem! hmm.
I don't mind it. Consider the gnome-terminal color settings:

Reply:
![image](https://user-images.githubusercontent.com/189190/97229445-a32af700-1795-11eb-8ee6-facc8747151e.png)

Reply:
I think a read-only textbox still requires me to observify the ColorScheme colors since the TextBox has to update itself when the color picker changes the color.  I could be rid of that if we go gnome and just show the colors and delete the TextBox altogether?

Reply:
Doesn't seem unreasonable to me!

Reply:
Should you somehow include the original colour there, so you can see what it has been changed from?  If not visually showing it, maybe include a ToolTip on each colour button with the name in it, like **Bright Red**

Reply:
Also how do the colour buttons re-arrange as the window is resized?  Do the Bright and Regular remain grouped, or does it change to a two column layout?

Reply:
Right now I've got it looking like this 
![image](https://user-images.githubusercontent.com/57155886/97237904-59e2a380-17a5-11eb-9b2d-b0bed48e999f.png)

Reply:
That looks reasonable, and clear.

I have been thinking, is there a plan to include an **Import** and **Export** button/option for the ColorScheme?  

I know there is the JSON settings page, but it would also be useful to support importing existing colour schemes like iTerm, JSON, or even raw TXT files.

Reply:
I love it. This is so great.

Martin, asking questions in PRs is a great way to make sure they get lost! Ask in 1564 so we remember to track it :smile: